scripts/release/mule/package/linux/amd64/deb/package.sh

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env bash
+# shellcheck disable=2038,2064
+
+set -ex
+
+echo
+date "+build_release begin PACKAGE DEB stage %Y%m%d_%H%M%S"
+echo
+
+OS_TYPE="$1"
+ARCH="$2"
+WORKDIR="$3"
+
+if [ -z "$OS_TYPE" ] || [ -z "$ARCH" ] || [ -z "$WORKDIR" ]; then
+    echo OS, ARCH and WORKDIR variables must be defined.
+    exit 1
+fi
+
+export REPO_DIR="$WORKDIR"
+export GOPATH="$REPO_DIR/go"
+export PATH="$GOPATH:/usr/local/go/bin:$PATH"
+
+mkdir -p "$REPO_DIR/tmp/node_pkgs/linux/amd64/pkg"
+export OUTDIR="$REPO_DIR/tmp/node_pkgs/$OS_TYPE/$ARCH/pkg"
+BRANCH=$("$REPO_DIR/scripts/compute_branch.sh")
+export BRANCH
+CHANNEL=$("$REPO_DIR/scripts/compute_branch_channel.sh" "$BRANCH")
+export CHANNEL
+export VARIATIONS="$OS_TYPE/$ARCH"
+
+PKG_NAME=$("$REPO_DIR/scripts/compute_package_name.sh" "${CHANNEL:-stable}")
+
+echo "Building debian package for '${OS} - ${ARCH}'"
+
+VER=$("$REPO_DIR/scripts/compute_build_number.sh" -f)
+
+if [ "${DEFAULTNETWORK}" = "" ]; then
+    DEFAULTNETWORK=$("$REPO_DIR/scripts/compute_branch_network.sh")
+fi
+DEFAULT_RELEASE_NETWORK=$("$REPO_DIR/scripts/compute_branch_release_network.sh" "${DEFAULTNETWORK}")
+
+PKG_ROOT=$(mktemp -d)
+trap "rm -rf $PKG_ROOT" 0
+
+mkdir -p "${PKG_ROOT}/usr/bin"
+
+if [ "${VARIATIONS}" = "" ]; then
+    # NOTE: keep in sync with installer/rpm/algorand.spec
+    bin_files=("algocfg" "algod" "algoh" "algokey" "carpenter" "catchupsrv" "ddconfig.sh" "diagcfg" "goal" "kmd" "msgpacktool" "node_exporter")
+fi
+
+for bin in "${bin_files[@]}"; do
```

Comment:
Could this be written so that it assumes these files are in the folders produced by ci-build? That way we can push the responsibility of producing artifacts to the build tasks. Also this may not work as expected where build artifacts are produced in a docker container and therefore won't be in the gopath of the host.

Reply:
I'll take a look at this.

Reply:
I've been thinking about this, and I think in order for this to happen some things need to be changed:

- The packaging is doing its own compilation of the binaries in the centos 7 image.  This is because we need to still support ubuntu 16.04 (although I think that may be deprecated soon), and the GNU C toolchain is too old on that distro so to be safe we're building on centos.

- Of course, the previous point could be made moot in two ways: the deprecation of support for ubuntu 16.04 or compiling our binaries in a way that doesn't have the external dependency on GNU C compiler.  I know there has been talk of this for a while, and I'm not sure what the state of it is.

- Also, how does the packaging "know" that the build artifacts are safe to be packaged?  What if there was a build that had been done a week ago but is stale now (perhaps you've already worked this out).

Once these issues are sorted, we could remove the build step from the packaging step and just pick up the previously-built assets, as you've suggested.

Reply:
The artifacts built by the mule build jobs also use centos 7. I think I actually started by copying the dockerfile you're using here

Reply:
https://github.com/algorand/go-algorand/blob/master/docker/build/cicd.Dockerfile

Reply:
Ok, that's good, I'll have that use those build artifacts then.